### PR TITLE
Init CI environment and test script

### PR DESCRIPTION
**Pull request by pxLi**
Initiate the CI setup and script,

A. Add CI-specific dockerfile, mostly was copied from [docker/Dockerfile.python](https://github.com/NVIDIA/spark-rapids-ml/blob/spark-cuml/docker/Dockerfile.python) w/ some changes,
1. use mamba to speed up conda installation
2. remove workdir copy of code base, and also remove pip installation since this should be covered in test script
3. pre-built image of CI in internal regular CI, if PR include change to ci/Dockerfile, it will build an temp image for test running within pre-merge

B. Add pre-merge jenkinsfile

C. Modify existing pom file to ref to released plugin jar for vulnerability scanning, as we do not publish plugin snapshots to mvn snapshots repo

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
@leewyang @wbo4958 @eordentlich please help review the CI script parts.

I will make some adjustments if any comments, thanks!

**Comment by pxLi** [comment]
convert to draft first to avoid accidental merge

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
internal jenkins job: rapids-ml_premerge-github, build ID:3

partial logs
```
[2023-03-16T11:06:06.623Z] + ./run_test.sh

[2023-03-16T11:06:07.998Z] All done! ✨ 🍰 ✨

[2023-03-16T11:06:07.998Z] 33 files would be left unchanged.

[2023-03-16T11:06:46.691Z] Success: no issues found in 33 source files

[2023-03-16T11:06:46.691Z] Formatting...

[2023-03-16T11:06:46.691Z] Type checking...

[2023-03-16T11:06:46.691Z] use --runslow to run all tests

[2023-03-16T11:06:51.942Z] ============================= test session starts ==============================

[2023-03-16T11:06:51.942Z] platform linux -- Python 3.8.10, pytest-7.2.2, pluggy-1.0.0

[2023-03-16T11:06:51.942Z] rootdir: /home/jenkins/agent/workspace/jenkins-rapids-ml_premerge-github-3

[2023-03-16T11:06:51.942Z] collected 224 items

[2023-03-16T11:06:51.942Z] 

[2023-03-16T11:07:25.880Z] tests/test_common_estimator.py ..                                        [  0%]

[2023-03-16T11:09:14.046Z] tests/test_kmeans.py ......ssssssssssss..                                [  9%]

[2023-03-16T11:12:40.032Z] tests/test_linear_model.py ..................sssssssssssssssssssssssssss [ 29%]

[2023-03-16T11:13:33.204Z] sssssssssssssssssssssssssssssssssssssssssssss.....                       [ 52%]

[2023-03-16T11:13:57.995Z] tests/test_nearest_neighbors.py ..ssssssssssss                           [ 58%]

[2023-03-16T11:16:15.182Z] tests/test_pca.py ........ssssssssssss..                                 [ 68%]

[2023-03-16T11:21:28.061Z] tests/test_random_forest.py ............................ssssssssssssssss [ 87%]

[2023-03-16T11:23:13.361Z] ssssssssssssssssssss....                                                 [ 98%]

[2023-03-16T11:23:17.538Z] tests/test_ucx.py .                                                      [ 99%]

[2023-03-16T11:23:17.538Z] tests/test_utils.py ..                                                   [100%]

[2023-03-16T11:23:17.538Z] 

[2023-03-16T11:23:17.538Z] =============================== warnings summary ===============================

[2023-03-16T11:23:17.538Z] tests/test_kmeans.py:258

[2023-03-16T11:23:17.538Z]   /home/jenkins/agent/workspace/jenkins-rapids-ml_premerge-github-3/tests/test_kmeans.py:258: PytestUnknownMarkWarning: Unknown pytest.mark.compat - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html

[2023-03-16T11:23:17.539Z]     @pytest.mark.compat

[2023-03-16T11:23:17.539Z] 

[2023-03-16T11:23:17.539Z] tests/test_linear_model.py:289

[2023-03-16T11:23:17.539Z]   /home/jenkins/agent/workspace/jenkins-rapids-ml_premerge-github-3/tests/test_linear_model.py:289: PytestUnknownMarkWarning: Unknown pytest.mark.compat - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html

[2023-03-16T11:23:17.539Z]     @pytest.mark.compat

[2023-03-16T11:23:17.539Z] 

[2023-03-16T11:23:17.539Z] tests/test_pca.py:284

[2023-03-16T11:23:17.539Z]   /home/jenkins/agent/workspace/jenkins-rapids-ml_premerge-github-3/tests/test_pca.py:284: PytestUnknownMarkWarning: Unknown pytest.mark.compat - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html

[2023-03-16T11:23:17.539Z]     @pytest.mark.compat

[2023-03-16T11:23:17.539Z] 

[2023-03-16T11:23:17.539Z] tests/test_random_forest.py:389

[2023-03-16T11:23:17.539Z]   /home/jenkins/agent/workspace/jenkins-rapids-ml_premerge-github-3/tests/test_random_forest.py:389: PytestUnknownMarkWarning: Unknown pytest.mark.compat - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html

[2023-03-16T11:23:17.539Z]     @pytest.mark.compat

[2023-03-16T11:23:17.539Z] 

[2023-03-16T11:23:17.539Z] tests/test_random_forest.py:516

[2023-03-16T11:23:17.539Z]   /home/jenkins/agent/workspace/jenkins-rapids-ml_premerge-github-3/tests/test_random_forest.py:516: PytestUnknownMarkWarning: Unknown pytest.mark.compat - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html

[2023-03-16T11:23:17.539Z]     @pytest.mark.compat

[2023-03-16T11:23:17.539Z] 

[2023-03-16T11:23:17.539Z] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

[2023-03-16T11:23:17.539Z] ============================= slowest 10 durations =============================

[2023-03-16T11:23:17.539Z] 47.53s call     tests/test_random_forest.py::test_random_forest_classifier_spark_compat[rf_types1]

[2023-03-16T11:23:17.539Z] 46.64s call     tests/test_random_forest.py::test_random_forest_regressor_spark_compat[rf_types1]

[2023-03-16T11:23:17.539Z] 41.10s call     tests/test_pca.py::test_pca_spark_compat[pca_types1]

[2023-03-16T11:23:17.539Z] 40.68s call     tests/test_linear_model.py::test_linear_regression_spark_compat[lr_types1]

[2023-03-16T11:23:17.539Z] 37.74s call     tests/test_kmeans.py::test_kmeans_spark_compat[kmeans_types1]

[2023-03-16T11:23:17.539Z] 31.77s call     tests/test_common_estimator.py::test_dummy

[2023-03-16T11:23:17.539Z] 21.24s call     tests/test_kmeans.py::test_kmeans_basic

[2023-03-16T11:23:17.539Z] 14.35s call     tests/test_random_forest.py::test_random_forest_basic[(100, 8)-float32-vector-(<class 'spark_rapids_ml.classification.RandomForestClassifier'>, <class 'spark_rapids_ml.classification.RandomForestClassificationModel'>, 2)]

[2023-03-16T11:23:17.539Z] 13.30s call     tests/test_linear_model.py::test_linear_regression_basic[0.0-(10, 2)-float32-vector]

[2023-03-16T11:23:17.539Z] 12.98s call     tests/test_linear_model.py::test_linear_regression_basic[0.0-(10, 2)-float32-array]

[2023-03-16T11:23:17.539Z] =========================== short test summary info ============================

[2023-03-16T11:23:17.539Z] SKIPPED [12] tests/test_kmeans.py:177: need --runslow option to run

[2023-03-16T11:23:17.539Z] SKIPPED [72] tests/test_linear_model.py:222: need --runslow option to run

[2023-03-16T11:23:17.539Z] SKIPPED [12] tests/test_nearest_neighbors.py:140: need --runslow option to run

[2023-03-16T11:23:17.539Z] SKIPPED [12] tests/test_pca.py:226: need --runslow option to run

[2023-03-16T11:23:17.539Z] SKIPPED [24] tests/test_random_forest.py:245: need --runslow option to run

[2023-03-16T11:23:17.539Z] SKIPPED [12] tests/test_random_forest.py:322: need --runslow option to run

[2023-03-16T11:23:17.539Z] =========== 80 passed, 144 skipped, 5 warnings in 989.93s (0:16:29) ============
```

```
[2023-03-16T11:25:30.480Z] ----------------------------------------------------------------------------------------------------

[2023-03-16T11:25:30.480Z] Spark Rapids ML LinearRegression training: 17.68 seconds

[2023-03-16T11:25:30.480Z] ----------------------------------------------------------------------------------------------------

[2023-03-16T11:25:30.480Z] ----------------------------------------------------------------------------------------------------

[2023-03-16T11:25:30.480Z] Spark ML LinearRegression transform: 12.14 seconds

[2023-03-16T11:25:30.480Z] ----------------------------------------------------------------------------------------------------

[2023-03-16T11:25:30.480Z] RMSE: 1.2260868811467996e-12, coefs l1: 433.9103006858376, coefs l2^2: 28733.097957666752, full_objective: 7.516445200601432e-25, intercept: -1.0058620603103918e-13

[2023-03-16T11:25:30.480Z] ----------------------------------------------------------------------------------------------------

[2023-03-16T11:25:30.481Z] benchmark time: : 35.66 seconds

[2023-03-16T11:25:30.481Z] ----------------------------------------------------------------------------------------------------
```

image

mark as Ready for review

File: ci/Jenkinsfile.premerge [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        """
+    }
+}
+
+void deleteDockerTempTag(String tag) {
+    if (!tag?.trim()) { // return if the tag is null or empty
+        return
+    }
+    sh "curl -u $ART_CREDS_USR:$ART_CREDS_PSW -XDELETE " +
+        "https://${ARTIFACTORY_NAME}/artifactory/sw-spark-docker-local/rapids-cuml/${tag} || true"
```
Comment (leewyang) [inline ci/Jenkinsfile.premerge:238]: same here.

File: ci/Jenkinsfile.premerge [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+import hudson.model.Run
+import jenkins.model.CauseOfInterruption.UserInterruption
+
+@Library(['blossom-lib']) _
+@Library('blossom-github-lib@master')
+import ipp.blossom.*
+
+def githubHelper // blossom github helper
+def TEMP_IMAGE_BUILD = true
+def IMAGE_PREMERGE = "${common.ARTIFACTORY_NAME}/sw-spark-docker/rapids-cuml:ubuntu20.04-cuda11.5.2-blossom"
```
Comment (leewyang) [inline ci/Jenkinsfile.premerge:33]: not sure how important/visible this image name is... but we've been trying to use `spark-rapids-ml` consistently throughout the codebase.

**Review by leewyang — APPROVED** [review]
Looks great! Just a minor question about the build image name.

File: ci/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Install conda
+ENV PATH="/root/miniconda3/bin:${PATH}"
+RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-py38_4.10.3-Linux-x86_64.sh \
+    && mkdir /root/.conda \
+    && bash Miniconda3-py38_4.10.3-Linux-x86_64.sh -b \
+    && rm -f Miniconda3-py38_4.10.3-Linux-x86_64.sh \
+    && conda init
+
+# install cuML
+ARG CUML_VER=22.12
```
Comment (wbo4958) [inline ci/Dockerfile:38]: Seems cuml 23.02 has been released for a while. Do we consider updating to 23.02?

File: ci/Jenkinsfile.premerge (hunk 2 above)
Comment (pxLi) [reply ci/Jenkinsfile.premerge]: sure, updated

File: ci/Dockerfile (hunk 3 above)
Comment (pxLi) [reply ci/Dockerfile:38]: This is the project-level decision, I think you can file an issue to track if needed.
If confirmed, we can bump versions in developer and CI Dockerfiles later

**Comment by pxLi** [comment]
build

**Review by wbo4958 — APPROVED** [review]

**Comment by wbo4958** [comment]
Great Job @pxLi. Thx

**Comment by pxLi** [comment]
Lets merge this now. People should be able to trigger build by comment `build` in PR,

I will keep it as optional check for some days, and convert it to required next week

**Pull request merged**